frontend/src/layouts/MainLayout/components/HeaderComponent.jsx

```diff
@@ -1,18 +1,30 @@
 import BCBox from '@/components/BCBox'
 import BCTypography from '@/components/BCTypography'
 import { useCurrentUser } from '@/hooks/useCurrentUser'
-import { numberFormatter } from '@/utils/formatters'
+import { useOrganizationBalance } from '@/hooks/useOrganization'
 import Icon from '@mui/material/Icon'
 import { useState } from 'react'
 import { useTranslation } from 'react-i18next'
 
 export const HeaderComponent = () => {
   const { t } = useTranslation()
   const { data, isFetched } = useCurrentUser()
-  const [showBalance, setShowBalance] = useState(false)
+  const [showBalance, setShowBalance] = useState(
+    !!+sessionStorage.getItem('showBalance') || false
```

Comment:
I haven't seen this !!+ notation before. Can you explain the purpose?

Reply:
The double bang will convert values to the Boolean, and the + notation will convert strings to numbers. So in this case because session storage can only store strings we are using '0' and '1' as false and true. We need to convert the string to a number and then use the double bang to convert to a Boolean. This way we aren't putting in conditionals in the render which should be cleaner